src/Components/ScreenerSurvey.js

```diff
@@ -29,12 +31,16 @@ function ScreenerSurvey({ surveyModel, version }) {
         }
     }, []);
 
-    function handleComplete(survey) {
+    async function handleComplete(survey) {
+        const uuid = uuidv4();
+
+        survey.data = { ...survey.data, uuid }; //asign locally created uuid
```

Comment:
Best practice to not reassign function arguments ([explanation](https://spin.atomicobject.com/2011/04/10/javascript-don-t-reassign-your-function-arguments/#:~:text=Regardless%20of%20the%20language%2C%20it,be%20a%20really%20bad%20idea.)). Instead we can drop `{ ...survey.data, uuid }` directly into `putSurveyResult`, or assign it to a new const.